packages/react/menu/src/Menu.tsx

```diff
@@ -361,6 +471,89 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
 
 MenuContent.displayName = CONTENT_NAME;
 
+/* -------------------------------------------------------------------------------------------------
+ * MenuSubContent
+ * -----------------------------------------------------------------------------------------------*/
+
+const SUB_CONTENT_NAME = 'MenuSubContent';
+
+type MenuSubContentOwnProps = Polymorphic.Merge<
+  Omit<
+    Polymorphic.OwnProps<typeof MenuContentImpl>,
+    'portalled' | 'disableOutsidePointerEvents' | 'disableOutsideScroll' | 'side' | 'trapFocus'
+  >,
+  {
+    /**
+     * Used to force mounting when more control is needed. Useful when
+     * controlling animation with React animation libraries.
+     */
+    forceMount?: true;
+  }
+>;
+type MenuSubContentPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof MenuContentImpl>,
+  MenuSubContentOwnProps
+>;
+
+const MenuSubContent = React.forwardRef((props, forwardedRef) => {
+  const { forceMount, dir, ...contentProps } = props;
+  const context = useMenuContext(SUB_CONTENT_NAME);
+  const subMenuContext = useMenuSubContext(SUB_CONTENT_NAME);
+  const direction = dir || context.dir;
+  const timerRef = React.useRef(0);
+
+  React.useEffect(() => {
+    return () => window.clearTimeout(timerRef.current);
+  }, []);
+
+  return (
+    <Presence present={forceMount || context.open}>
+      <CollectionSlot>
+        <MenuContentImpl
+          id={subMenuContext.contentId}
+          align="start"
+          {...contentProps}
+          side={direction === 'ltr' ? 'right' : 'left'}
+          portalled
+          disableOutsidePointerEvents={false}
+          disableOutsideScroll={false}
+          trapFocus={false}
+          ref={composeRefs(forwardedRef, subMenuContext.contentRef)}
+          dir={direction}
+          onKeyDown={composeEventHandlers(contentProps.onKeyDown, (event) => {
+            const tabNext = event.key === 'Tab' && !event.shiftKey;
+            if (SUB_CLOSE_KEYS[direction].includes(event.key) || tabNext) {
+              // close a single level unless it's a forward tab, in which case close the entire stack
```

Comment:
Curious where you got the idea for this behaviour from? I'm testing a nested menu on MacOS and it seems `shift+tab` and `tab` always closes the entire stack.